utils/util_test.go

```diff
@@ -27,3 +28,35 @@ func TestBitsBytesConvert(t *testing.T) {
 		}
 	}
 }
+
+func TestIntToBytes(t *testing.T) {
+	numInt := 42
+	b := IntToBytes(numInt)
+	if int(binary.LittleEndian.Uint32(b)) != numInt {
+		t.Fatal("Conversion to bytes looks wrong!")
+	}
+	// TODO It is possible to call `IntToBytes` with a signed int < 0
+	// isn't that problematic?
```

Comment:
Refactoring ``level`` to be unsigned makes a lot of sense to me. The only scenario in which level could decrease is when the tree is re-randomized after the VRF key is changed. But this should still work with an unsigned ``level``.